references/recognition/train_pytorch.py

```diff
@@ -190,88 +122,123 @@ def evaluate(model, val_loader, batch_transforms, val_metric, amp=False, log=Non
 
 
 def main(args):
+    """
+    Args:
+        rank (int): device id to put the model on
+        world_size (int): number of processes participating in the job
+        args: other arguments passed through the CLI
+    """
+    # Detect distributed setup
+    world_size = int(os.environ.get("WORLD_SIZE", 1))
+    distributed = world_size > 1
+
+    # GPU setup
+    if distributed:
+        rank = int(os.environ.get("LOCAL_RANK", 0))
```

Comment:
rank = dist.get_rank()

this should be only != 0 if multi gpus are available right ?